### ISSUE TITLE
RegionCLIP: Region-based Language-Image Pretraining

### ISSUE THREAD

**Issue by zhuoxiao-chen** [zhuoxiao-chen/please_read_paper_every_day#35]
### Paper Link
https://openaccess.thecvf.com/content/CVPR2022/papers/Zhong_RegionCLIP_Region-Based_Language-Image_Pretraining_CVPR_2022_paper.pdf

### Code Link
https://github.com/microsoft/RegionCLIP

## Motivation
<img width="500" alt="image" src="https://user-images.githubusercontent.com/50613939/235827238-824d7bc0-04f0-4333-a642-cdb5fdce51a3.png">

- Construct a simple R-CNN, that crops candidate object regions from an input image, and applies the CLIP model for detection by matching visual features of cropped regions to text embeddings of object categories.
- Scores from CLIP often fail to capture the **localization quality** (Fig. 1a).
- Even with **groundtruth object boxes**, classification accuracy using CLIP drops significantly from 60% on ImageNet to 19% on LVIS, with a similar number of classes (Fig. 1b).

**The key idea is to explicitly align image regions and text tokens during pretraining.**

### Challenges

1. the fine-grained alignment between image regions and text tokens is **not available** in image-text pairs and expensive to annotate.
2. the text description of an image is often **incomplete**, i.e. many image regions are not described by the text.

### Solution
Bootstrap from a pretrained vision-language model to align image regions and text tokens, and to fill in the missing region descriptions, as illustrated in Fig. 1c.

## Methodology

<img width="1552" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/85b78d8e-5eab-4be8-9aa2-48f8ebd06e3e">

### Visual and Semantic Region Representation

**Visual region representation.** By default, we use RPN pretrained on human-annotated object bounding boxes without object labels. Given the proposed regions, the visual representation $v_i$  of region $r$ i is extracted from our visual encoder $V$ with a feature pooling method.

**Semantic region representation.** We first build a large pool of concepts to exhaustively cover regional concepts. As shown at the bottom of Fig. 2, we create **a pool of object concepts** which are parsed from text corpus (e.g., the image descriptions collected from the Internet), by using off-the-shelf language parsers.

Semantic representations for regions are created by two steps:
1. a short sentence for each concept is created by filling it to prompt templates (e.g., prompts of CLIP), e.g., the “kite” concept is converted to “A photo of a kite”;
2. the resulting text descriptions are further encoded into semantic representations by using the pretrained language encoder $L$.

### Visual-Semantic Alignment for Regions

We leverage a teacher visual encoder $V_t$ to connect image regions and our created texts:

<img width="200" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/07ad3522-22f9-44c4-a308-25fcf5cffabe">

Finally, we obtain a pseudo label for each region, forming the pairs of 
$\{v_i, l_m\}.$

### Pretraining

**Contrastive Loss**

<img width="400" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/08fa7003-1701-40bb-afa7-2e16435ec784">

**Distillation Loss**

<img width="400" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/683227be-3685-4f94-a50a-4d5386667119">

**Final Loss**

<img width="400" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/e2b9d9c6-f846-453d-a0e4-59f6d4073e5c">

**Zero-shot Inference**

For example, given **region proposals** from RPN, region representations extracted from our visual encoder can be used to **match the embeddings of target object concepts**, and thus recognize the concepts within local image regions, thereby enabling zero-shot inference for object detection.

### Transfer Learning for Object Detection

When strong supervision for image regions is available (e.g., **the humanannotated detection labels**), our visual encoder can be **further fine-tuned by replacing the region descriptions with human annotations**, as shown in Panel 3 of Fig. 2.

### Open-Vocabulary Object Detection

<img width="1491" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/22684ab7-74d1-447c-95d7-6092031fe155">

### Fully Supervised Object Detection
<img width="1575" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/2165c794-785f-47f6-9f12-98fd5cd9824f">

### Zero-shot Inference with Ground-truth (GT) Boxes
<img width="500" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/8e019511-c7dc-467b-be72-3c7e2762c04c">

### Visualization

<img width="1768" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/c0bc8ad9-616c-4728-ba03-f08e6247b0c1">

<img width="527" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/3c79d2c6-a343-45f2-b98b-4d6bf44ac665">
